features/frameworks/greentea-client/source/greentea_test_env.cpp

```diff
@@ -562,7 +560,20 @@ enum Token {
  *
  */
 extern "C" int greentea_getc() {
-    return greentea_serial->getc();
+    uint8_t c;
+    read(STDERR_FILENO, &c, 1);
+    return c;
+}
+
+
+/**
+ * \brief Write character from stream of data
+ *
+ * \return The number of bytes written
+ *
+ */
+extern "C" int greentea_putc(int c) {
+    return write(STDERR_FILENO, &c, 1);
```

Comment:
This isn't strictly portable - it assumes little-endianness.

Bit nicer to assign `c` to a `uint8_t` first. Would align with `getc`.

Reply:
Please review [this](https://github.com/ARMmbed/mbed-os/compare/10b2d1c8aa95b39231cb96777f816895ce9538bb..72e390ca702f6528c2c52c576808d7a4decbbfd5).